agents/ecloud/fence_ecloud.py

```diff
@@ -0,0 +1,165 @@
+#!@PYTHON@ -tt
+#
+# Fence agent for eCloud and eCloud VPC
+# https://www.ans.co.uk/cloud-and-infrastructure/ecloud/
+#
+# Copyright (c) 2022 ANS Group Limited
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.  If not, see
+# <http://www.gnu.org/licenses/>.
+
+import sys
+import time
+import atexit
+import logging
+import requests
+sys.path.append("@FENCEAGENTSLIBDIR@")
+from fencing import *
+from fencing import run_delay, fail_usage, fail, EC_TIMED_OUT
+
+API_BASE = "https://api.ukfast.io/ecloud"
+API_MONITOR = API_BASE + "/ping"
+API_VPC_INSTANCE_DATA = API_BASE + "/v2/instances/:ID"
+API_VPC_POWER_ON = API_BASE + "/v2/instances/:ID/power-on"
+API_VPC_POWER_OFF = API_BASE + "/v2/instances/:ID/power-off"
+API_V1_INSTANCE_DATA = API_BASE + "/v1/vms/:ID"
+API_V1_POWER_ON = API_BASE + "/v1/vms/:ID/power-on"
+API_V1_POWER_OFF = API_BASE + "/v1/vms/:ID/power-off"
+
+
+def docs(options):
```

Comment:
You might want to use another name for the function here, as using docs for both the function as the dictionary might be a bit confusing.